lib/gtfs.ex

```diff
@@ -0,0 +1,165 @@
+defmodule Gtfs do
+  use GenServer
+  require Logger
+
+  @type t :: %__MODULE__{
+          routes: [Gtfs.Route.t()],
+          stop_times: [Gtfs.StopTime.t()],
+          stops: [Gtfs.Stop.t()],
+          trips: [Gtfs.Trip.t()]
+        }
+
+  @enforce_keys [
+    :routes,
+    :stop_times,
+    :stops,
+    :trips
+  ]
+
+  defstruct [
+    :routes,
+    :stop_times,
+    :stops,
+    :trips
+  ]
+
+  @type state :: t
+
+  @type query_opts :: %{
+          optional(:server) => GenServer.server()
+        }
+
+  @type files :: %{optional(String.t()) => binary()}
+
+  @type files_source :: {:url, String.t()} | {:mocked_files, mocked_files()}
+
+  @typedoc """
+  For mocking tests
+  E.g.
+  %{
+    "stops.txt" => [
+      "stop_id,stop_name",
+      "place-sstat,South Station",
+    ]
+  ]}
+  """
+  @type mocked_files :: %{optional(String.t()) => [binary()]}
+
+  # Client functions
+
+  @spec start_link(String.t()) :: GenServer.on_start()
+  def start_link(url) do
+    GenServer.start_link(__MODULE__, {:url, url}, name: __MODULE__)
+  end
+
+  @spec start_mocked(mocked_files) :: GenServer.on_start()
+  def start_mocked(mocked_files) do
+    GenServer.start_link(__MODULE__, {:mocked_files, mocked_files})
```

Comment:
Seems like we aren't using it right now, but would it make sense to include `name: __MODULE__` here too as in line 52 for consistency?

Reply:
The goal here was to be able to have multiple mocked genservers running independently. They would be accessed by their id, and the `name: __MODULE__` would be reserved for the genserver with real data.

Reply:
Ah right, of course.